rero_ils/modules/patron_transaction_events/api.py

```diff
@@ -128,16 +128,25 @@ def create_event_from_patron_transaction(
 
     def update_parent_patron_transaction(self):
         """Update parent patron transaction amount and status."""
+        # NOTE :
+        #   due to bit representation of float number
+        #   (https://en.wikipedia.org/wiki/IEEE_754), the arithmetic operation
+        #   with float can cause some strange behavior
+        #     >>> 10 - 9.54
+        #     0.46000000000000085
+        #   To solve this problem in our case, as we keep only 2 decimal
+        #   digits, we can multiply amounts by 100, cast result as integer,
+        #   do operation with these values, and (at the end) divide the result
+        #   by 100.
         patron_transaction = self.patron_transaction()
-        total_amount = patron_transaction.get('total_amount')
+        total_amount = int(patron_transaction.get('total_amount') * 100)
```

Comment:
My understanding is the `int` removes the decimal part of the value. is this working as expected?

Reply:
exactly ! According with PO, we only keep 2 decimals. So multiply by 100 and cast to int do the works (and divide at the end)